### PR TITLE
Sync the code documentation from the README to the source files

### LINKED ISSUE #300: Copy the class descriptions from the README into code class comments
I'll take this.

### PR DESCRIPTION
**Pull request by oliverklee**
This should help readers understand the code.

Also drop outdated documentation and polish the existing prose.

Fixes #300

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -69,29 +69,27 @@ The resulting data structure consists mainly of five basic types: `CSSList`, `Ru
...
 
 #### RuleSet
 
 `RuleSet` is a container for individual rules. The most common form of a rule set is one constrained by a selector. The following concrete subtypes exist:
 
-* `AtRuleSet` – for generic at-rules which do not match the ones specifically mentioned like `@import`, `@charset` or `@media`. A common example for this is `@font-face`.
+* `AtRuleSet` – for generic at-rules for generic at-rules which are not covered by specific classes, i.e., not `@import`, `@charset` or `@media`. A common example for this is `@font-face`.
 * `DeclarationBlock` – a `RuleSet` constrained by a `Selector`; contains an array of selector objects (comma-separated in the CSS) as well as the rules to be applied to the matching elements.
 
+///////////////// continue here
```
Comment (sabberworm) [inline README.md:85]: ?

Comment (oliverklee) [reply README.md:85]: Oops. Thanks for spotting this!

Comment (oliverklee) [reply README.md]: Done.

**Comment by sabberworm** [comment]
Thanks a lot!

**Pull request merged**